### PR TITLE
Revert "Revert "[94X] L1 no hack conditions - cleaned""

### PR DESCRIPTION
**Pull request by rekovic**
Reverting the revert of No Hack Conditions for L1T, by reverting commit cddc18fde9d0fbbf4e0639d7d76926020bddfb45.

Plus, with the last commit fa19aae, it reverts for AlCa run2_mc_hi to use the correct tag `93X_mcRun2_HeavyIon_v0` which contains payloads for Stage1, needed for previously failing wf 145.0.

### CODE REVIEW

**Comment by rekovic** [comment]
please test

**Comment by rekovic** [comment]
please test

File: L1Trigger/L1TMuon/plugins/L1TMuonProducer.cc [hunk 1]
```diff
@@ -21,6 +21,7 @@
 // system include files
 #include <memory>
 #include <fstream>
+#include <sstream>
```
Comment (thomreis) [inline L1Trigger/L1TMuon/plugins/L1TMuonProducer.cc]: Hi @rekovic ,
this include could be removed as well.

Comment (rekovic) [reply L1Trigger/L1TMuon/plugins/L1TMuonProducer.cc]: Yes. Thx.

Comment (thomreis) [reply L1Trigger/L1TMuon/plugins/L1TMuonProducer.cc]: I meant the sstream, actually.

**Comment by davidlange6** [comment]
please test

File: L1Trigger/L1TMuon/plugins/L1TMuonProducer.cc (hunk 1 above)
Comment (thomreis) [reply L1Trigger/L1TMuon/plugins/L1TMuonProducer.cc]: fstream is needed for std::ofstream m_debugOut;

File: L1Trigger/L1TMuon/plugins/L1TMuonProducer.cc [hunk 2]
```diff
@@ -20,7 +20,7 @@
 
 // system include files
 #include <memory>
-#include <fstream>
+#include <sstream>
```
Comment (thomreis) [inline L1Trigger/L1TMuon/plugins/L1TMuonProducer.cc]: This should stay fstream. sstream is not needed but fstream is needed for std::ofstream m_debugOut; in line 120.

**Comment by rekovic** [comment]
please test

**Comment by davidlange6** [comment]
> On Oct 19, 2017, at 8:21 PM, cmsbuild <notifications@github.com> wrote:
> 
> -1
> 
> Tested at: adb254c
> 
> The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic:
> e2f528f
> You can see more details here:
> https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-20957/23883/git-log-recent-commits
> https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-20957/23883/git-merge-result
> 
> You can see the results of the tests here:
> https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-20957/23883/summary.html
> 
> I found follow errors while testing this PR
> 
> Failed tests: AddOn
> 
> 	• AddOn:
> I found errors in the following addon tests:
> 
> cmsDriver.py TTbar_Tauola_13TeV_TuneCUETP8M1_cfi -s GEN,SIM,DIGI,L1,DIGI2RAW --mc --scenario=HeavyIons -n 10 --conditions auto:run2_mc_HIon --relval 9000,50 --datatier "GEN-SIM-RAW" --eventcontent RAWSIM --customise=HLTrigger/Configuration/CustomConfigs.L1T --era Run2_2016,Run2_HI --fileout file:RelVal_Raw_HIon_MC.root : FAILED - time: date Thu Oct 19 19:41:57 2017-date Thu Oct 19 19:38:49 2017 s - exit: 34304
> 

This scenario seems not to make any sense to me - @Martin-Grunewald what is the idea here? There is no such data.

**Comment by Martin-Grunewald** [comment]
@davidlange6 
What part? I am using TTbar because true high-pt HI would take too long.

**Comment by davidlange6** [comment]
There is no hi data in 2016. It was 2015 ( which matters for l1)



On 20 Oct 2017, at 06:31, Martin Grunewald <notifications@github.com<mailto:notifications@github.com>> wrote:


@davidlange6<https://github.com/davidlange6>
What part? I am using TTbar because true high-pt HI would take too long.

**Comment by davidlange6** [comment]
"hi" means PbPb - there is a separate pPb GT for 2016 (auto:run2_mc_pa).

**Comment by Martin-Grunewald** [comment]
Yes, but the addOn HIon step runs L1Menu_CollisionsHeavyIons2015_v5_uGT_xml which is the uGT version of the HIon L1T menu.

**Comment by davidlange6** [comment]
ok - i let you and L1 agree on what to do (today)

the error is

%MSG-i ThreadStreamSetup:  (NoModuleName) 19-Oct-2017 19:38:59 CEST pre-events
setting # threads 4
setting # streams 4
%MSG
%MSG-e L1TCaloLayer1FetchLUTs:  L1TCaloLayer1:simCaloStage2Layer1Digis@beginRun  19-Oct-2017 19:40:29 CEST Run: 1
caloLSB (caloParams.towerLsbSum()) != 0.5, actually = 0
%MSG
%MSG-e L1TCaloLayer1FetchLUTs:  L1TCaloLayer1:simCaloStage2Layer1Digis@beginRun  19-Oct-2017 19:40:29 CEST Run: 1
Layer 1 calibrations requested (useCalib = True) but there are missing scale factors in CaloParams!  Please check conditions setup.
%MSG
%MSG-e L1TCaloLayer1:  L1TCaloLayer1:simCaloStage2Layer1Digis@beginRun  19-Oct-2017 19:40:29 CEST Run: 1
L1TCaloLayer1::beginRun: failed to fetch LUTS - using unity
%MSG
----- Begin Fatal Exception 19-Oct-2017 19:40:56 CEST-----------------------
An exception of category 'NoRecord' occurred while
   [0] Processing global begin Run run: 1
   [1] Calling method for module L1TMuonOverlapTrackProducer/'simOmtfDigis'
Exception Message:
No "L1TMuonOverlapParamsRcd" record found in the EventSetup for synchronization value
Run: 1 LuminosityBlock: 0 Event: 0 Time: 1
 Please add an ESSource or ESProducer that delivers such a record.
----- End Fatal Exception -------------------------------------------------

**Comment by rekovic** [comment]
@Martin-Grunewald 
You can run the menu `L1Menu_CollisionsHeavyIons2015_v5_uGT_xml `, that is all fine. 

We need to remove `Run2_2016` part from `--era Run2_2016,Run2_HI` from the definition of this AddOn workflow, as the two eras have conflicting L1T modifiers.  The former defines L1T as  Stage2, and the latter defines it as Stage2.   It is the latter that we want. 
 
`Run2_2016 = cms.ModifierChain(run2_common, run2_25ns_specific, stage2L1Trigger, ctpps_2016)`

and.

`Run2_HI = cms.ModifierChain(run2_common, run2_HI_specific, stage1L1Trigger)`

I can make this change in this PR.

**Comment by Martin-Grunewald** [comment]
OK, let's try that!

**Comment by rekovic** [comment]
Successfully tested:

`cmsDriver.py TTbar_Tauola_13TeV_TuneCUETP8M1_cfi -s GEN,SIM,DIGI,L1,DIGI2RAW --mc --scenario=HeavyIons -n 10 --conditions auto:run2_mc_HIon --relval 9000,50 --datatier "GEN-SIM-RAW" --eventcontent RAWSIM --customise=HLTrigger/Configuration/CustomConfigs.L1T --era Run2_HI --fileout file:RelVal_Raw_HIon_MC.root`

**Comment by Martin-Grunewald** [comment]
@rekovic 
It occurs 4 times in the addOnTests.py file (2x data, 2x MC) - I assume all should be updated for consistency.

**Comment by rekovic** [comment]
@Martin-Grunewald 
FYI, There is one more AddOn test wf, which runs on HI 2015 data, which uses contradicting L1T definitions.
`                    'hlt_data_HIon' : ['cmsDriver.py RelVal -s L1REPACK:Full2015Data --data --scenario=HeavyIons -n 10 --conditions auto:run2_hlt_HIon --relval 9000,50 --datatier "RAW" --eventcontent RAW --customise=HLTrigger/Configuration/CustomConfigs.L1T --era Run2_2016,Run2_HI --fileout file:RelVal_Raw_HIon_DATA.root --filein /store/hidata/HIRun2015/HIHardProbes/RAW-RECO/HighPtJet-PromptReco-v1/000/263/689/00000/1802CD9A-DDB8-E511-9CF9-02163E0138CA.root'`

This one does not crash because L1T Stage2 is defined and `L1REPACK:Full2015Data` runs fine. 
Maybe this is exactly what we really want to achieve with this test, to run new L1Trigger on old HI data, I don't know, but I thought I should point it out.

I will procede with making the change for the AddOn test wf `'hlt_mc_HIon'`.

**Comment by rekovic** [comment]
@Martin-Grunewald 
> It occurs 4 times in the addOnTests.py file (2x data, 2x MC) - I assume all should be updated for consistency.

No, L1REPACK:Full2015Data would not work for Stage1 L1T.  It is designed for Stage2 L1T only.

```
from Configuration.Eras.Modifier_stage2L1Trigger_cff import stage2L1Trigger
if not (stage2L1Trigger.isChosen()):
    print "L1T WARN:  L1REPACK:Full2015Data only supports Stage 2 eras for now."
    print "L1T WARN:  Use a legacy version of L1REPACK for now."
```

So for now, I will leave the wf `'hlt_data_HIon'` as is.

**Comment by rekovic** [comment]
please test

**Comment by rekovic** [comment]
please test

**Comment by rekovic** [comment]
please test

**Comment by rekovic** [comment]
+1

**Comment by davidlange6** [comment]
merge

**Pull request merged**